.ci/scripts/release/prepare.sh

```diff
@@ -1,6 +1,10 @@
 #!/bin/bash -xue
 
-# configure Jenkins GitHub user
+# remove origin and use GitHub App (reflected on filesystem and globally active)
+git remote remove origin
+git remote add origin https://${GITHUB_TOKEN_USR}:${GITHUB_TOKEN_PSW}@github.com/camunda-cloud/zeebe.git
```

Comment:
From what I could see nothing is pushed to Git in the distribution scripts/Jenkinsfile.

Currently, the `camunda-jenkins` user is used for anything in regards to releasing (withCredentials - ssh key), but the user is essentially deprecated since GitHub Apps were introduced in the meantime and the user doesn't belong to the `camunda-cloud` org due to missing 2FA keys.

The alternative is to use GitHub Apps, which creates a signed token and allows logging in/pushing etc. via that signed token. Since we don't have an ssh key anymore we use the token for the release process.
Removing the origin + recreating it with the credentials allows the pipeline to use it anywhere throughout as long as it is within the repository, meaning the maven release can update its version, as well as the go container, update its client.

Instead of doing it this way one could change all `git push` to `git push https://user:pass@github.com/camunda-cloud/zeebe branch`.

Reply:
sadly there is no way to properly test it since the repository is moved to another org.